### ISSUE TITLE
Adding plugins from pom.xml fails

### ISSUE THREAD

**Issue by Xaseron** [jenkinsci/custom-war-packager#95]
I tried to generate the war for https://github.com/oleg-nenashev/ci.jenkins.io-runner but it failed:
```
Exception in thread "main" java.io.IOException: CasC section is declared, but CasC plugin is not declared in the plugins list
	at io.jenkins.tools.warpackager.lib.impl.Builder.verifyConfig(Builder.java:66)
	at io.jenkins.tools.warpackager.lib.impl.Builder.build(Builder.java:97)
	at io.jenkins.tools.warpackager.cli.Main.main(Main.java:53)
make: *** [Makefile:38: build] Error 1
```

i tracked the error down to the missing dependencies. Every plugin is a jar file in `tmp/hpiDownloads/dependencies.txt`
e.g. :
```
The following files have been resolved:
   org.jenkins-ci.plugins.workflow:workflow-job:jar:2.32:compile
   org.apache.commons:commons-csv:jar:1.5:compile
   org.apache.maven:maven-plugin-api:jar:3.1.0:compile
   org.jenkins-ci.ui:handlebars:jar:1.1:compile
   org.jenkins-ci.plugins.workflow:workflow-durable-task-step:jar:2.22:compile
...
```

i've wrote a small patch that doesn't check for the defined dependencies to be hpi files and it is working for me.

```diff
Index: custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/config/Config.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/config/Config.java	(date 1557242355000)
+++ custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/config/Config.java	(date 1557308159000)
@@ -207,9 +207,7 @@
 
         List<DependencyInfo> deps = helper.listDependenciesFromPom(tmpDir, pom, destination);
 
-        for (DependencyInfo dep : deps) {
-            processMavenDep(helper, tmpDir, dep, plugins);
-        }
+        plugins.addAll(deps);
 
         if (!pomIgnoreRoot) {
             // Add the artifact itself, no validation as we assume the pom is from a plugin
```

**Comment by oleg-nenashev** [comment]
working on it

**Comment by oleg-nenashev** [comment]
https://github.com/jenkinsci/custom-war-packager/pull/99 should fix it

**Comment by Xaseron** [comment]
I get a lot of errors downloading the plugins.  I have deleted the proxy setting from the log.
e.g.:
```
java \
    -jar ../../custom-war-packager-cli/target/custom-war-packager-cli-1.7-SNAPSHOT-jar-with-dependencies.jar \
    -configPath packager-config.yml -version 256.0-test
Picked up _JAVA_OPTIONS: -Dhttp.proxyHost=companyproxy
Picked up _JAVA_OPTIONS: -Dhttp.proxyHost=companyproxy
May 20, 2019 12:05:56 PM io.jenkins.tools.warpackager.lib.util.MavenHelper artifactExistsInLocalCache
INFO: Checking /home/mcwr/.m2/repository/org/jenkins-ci/plugins/workflow/workflow-job/2.32/workflow-job-2.32.hpi
May 20, 2019 12:05:56 PM io.jenkins.tools.warpackager.lib.util.MavenHelper artifactExistsInLocalCache
INFO: Checking /home/mcwr/.m2/repository/org/apache/commons/commons-csv/1.5/commons-csv-1.5.hpi
Picked up _JAVA_OPTIONS: -Dhttp.proxyHost=companyproxy
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:2.8:get (default-cli) on project standalone-pom: Couldn't download artifact: Could not find artifact org.apache.commons:commons-csv:hpi:1.5 in central (https://repo.maven.apache.org/maven2)
[ERROR] 
[ERROR] Try downloading the file manually from the project website.
[ERROR] 
[ERROR] Then, install it using the command: 
[ERROR]     mvn install:install-file -DgroupId=org.apache.commons -DartifactId=commons-csv -Dversion=1.5 -Dpackaging=hpi -Dfile=/path/to/file
[ERROR] 
[ERROR] Alternatively, if you host your own repository you can deploy the file there: 
[ERROR]     mvn deploy:deploy-file -DgroupId=org.apache.commons -DartifactId=commons-csv -Dversion=1.5 -Dpackaging=hpi -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR] 
[ERROR] 
[ERROR]   org.apache.commons:commons-csv:hpi:1.5
[ERROR] 
[ERROR] from the specified remote repositories:
[ERROR]   central (https://repo.maven.apache.org/maven2, releases=true, snapshots=false)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
May 20, 2019 12:06:00 PM io.jenkins.tools.warpackager.lib.util.MavenHelper artifactExists
INFO: Could not download org.apache.commons:commons-csv:1.5, assuming it is not an HPI and creating /home/mcwr/.m2/cwp_non_hpi_cache/org/apache/commons/commons-csv/1.5/commons-csv-1.5.hpi to remember the assumption.
May 20, 2019 12:06:00 PM io.jenkins.tools.warpackager.lib.config.Config processMavenDep
INFO: Skipping dependency, not an HPI file: org.apache.commons:commons-csv:1.5
May 20, 2019 12:06:00 PM io.jenkins.tools.warpackager.lib.util.MavenHelper artifactExistsInLocalCache
```

**Comment by oleg-nenashev** [comment]
This log errors are actually expected.

```
May 20, 2019 12:06:00 PM io.jenkins.tools.warpackager.lib.util.MavenHelper artifactExists
INFO: Could not download org.apache.commons:commons-csv:1.5, assuming it is not an HPI and creating /home/mcwr/.m2/cwp_non_hpi_cache/org/apache/commons/commons-csv/1.5/commons-csv-1.5.hpi to remember the assumption.
May 20, 2019 12:06:00 PM io.jenkins.tools.warpackager.lib.config.Config processMavenDep
```

This is a downside of the current caching logic which we need to fix by just using update center as an information source

**Comment by oleg-nenashev** [comment]
Finally released in 1.7 https://github.com/jenkinsci/custom-war-packager/releases/tag/parent-pom-1.7
